### PR TITLE
rawagner_Implement click() method within TreeItemHandler

### PR DESCRIPTION
**Pull request by vpakan**

### CODE REVIEW

File: plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/TreeItemHandler.java [hunk 1]
```diff
@@ -381,6 +382,60 @@ public void run() {
...
+	 * Creates Mouse Event for TreeItem
+	 * @param swtTreeItem
+	 * @param x
+	 * @param y
+	 * @param button
+	 * @param stateMask
+	 * @param count
+	 * @return
+	 */
+	private Event createMouseEvent(TreeItem swtTreeItem, int x, int y, int button, int stateMask, int count) {
```
Comment (rawagner) [inline plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/TreeItemHandler.java]: Might be safer to use TreeHandler#createEventForTree at first and then add other needed info. 
createEventForTree specifies other event attributes such as type and item which might be missing depending on listener implementation

**Comment by rawagner** [comment]
Can you also add test for this ?

File: plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/TreeItemHandler.java (hunk 1 above)
Comment (vpakan) [inline plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/TreeItemHandler.java]: I don't understand. This method is creating mouse event for tree item not event for tree. What would be safer?

**Comment by vpakan** [comment]
Any ideas about test scenario? I don't know how to validate if mouse click on tree item actually happened. Defining listener and evaluating event data within listener doesn't make sense.

It would be tested in JBT integration tests using more complex test scenario

File: plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/TreeItemHandler.java (hunk 1 above)
Comment (rawagner) [inline plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/TreeItemHandler.java]: I didn't notice that event.widget has to be treeItem. Then add event.item and event.type to this method

**Comment by rawagner** [comment]
Test - ShellMenu File->New->Other... click on any treeItem and check if it was selected would be enough to have at least something.

**Comment by vpakan** [comment]
Ready for push

**Pull request closed without merging**